### PR TITLE
Change startup commands for bun and deno

### PR DESCRIPTION
**Pull request by jim-king-2000**
See elysiajs/elysia#64.

### CODE REVIEW

File: javascript/config.yaml [hunk 1]
```diff
@@ -27,9 +27,9 @@ language:
         - npm install
       command: pm2-runtime start app.js -i $(nproc)
     deno:
-      command: deno run --allow-net --allow-read --allow-env app.ts
+      command: pm2 start app.ts --interpreter="deno" --interpreter-args="run --allow-net"
     bun:
-      command: bun run app.ts
+      command: pm2-runtime start "NODE_ENV=production bun app.ts" -i max
```
Comment (waghanza) [inline javascript/config.yaml:32]:
```suggestion
   deno:
      deps:
        - npm
      bootstrap:
        - npm install -g pm2
      command: pm2-runtime start "NODE_ENV=production deno run --allow-net --allow-read --allow-env app.ts" -i max
    bun:
      deps:
        - npm
      bootstrap:
        - npm install -g pm2
      command: pm2-runtime start "NODE_ENV=production bun app.ts" -i max
```

**Review by waghanza — CHANGES_REQUESTED** [review]
`pm2` is not installed in containers.

You will need to add
```
{{#deps}}
  RUN apk add {{{.}}}
{{/deps}}

{{#bootstrap}}
  RUN {{{.}}}
{{/bootstrap}}
```
after `WORKDIR` in `deno.Dockerfile`

and `bun install` prior to `CMD` in `bun.Dockerfile`

File: javascript/config.yaml (hunk 1 above)
Comment (jim-king-2000) [reply javascript/config.yaml]: > `pm2` is not installed in containers.


Change it to `pm2-runtime`.


> You will need to add
{{#deps}}
  RUN apk add {{{.}}}
{{/deps}}

> {{#bootstrap}}
  RUN {{{.}}}
{{/bootstrap}}

> after `WORKDIR` in `deno.Dockerfile`


Done.


> and bun install prior to CMD in bun.Dockerfile

Done.

**Comment by jim-king-2000** [comment]
> `pm2` is not installed in containers.


Change it to `pm2-runtime`.


> You will need to add
{{#deps}}
  RUN apk add {{{.}}}
{{/deps}}

> {{#bootstrap}}
  RUN {{{.}}}
{{/bootstrap}}

> after `WORKDIR` in `deno.Dockerfile`


Done.


> and bun install prior to CMD in bun.Dockerfile

Done.

**Comment by jim-king-2000** [comment]
Still some errors. Any solution?

File: javascript/config.yaml [hunk 2]
```diff
@@ -27,9 +27,9 @@ language:
         - npm install
       command: pm2-runtime start app.js -i $(nproc)
     deno:
-      command: deno run --allow-net --allow-read --allow-env app.ts
+      command: pm2-runtime start app.ts --interpreter="deno" --interpreter-args="run --allow-net"
```
Comment (waghanza) [inline javascript/config.yaml:30]:
```suggestion
      command: pm2 start app.ts --interpreter="deno" --interpreter-args="run --allow-net --allow-read --allow-env" -i max &
```

**Review by waghanza — COMMENTED** [review]
Sorry, missing some configuration options
```yaml
     deps:
        - npm
      bootstrap:
        -  npm --location=global install pm2
```
for `bun` and `deno`

File: javascript/config.yaml [hunk 3]
```diff
@@ -27,9 +27,9 @@ language:
         - npm install
       command: pm2-runtime start app.js -i $(nproc)
     deno:
```
Comment (waghanza) [inline javascript/config.yaml:29]:
```suggestion
    deno:
      deps:
        - npm
      bootstrap:
        -  npm --location=global install pm2
```

File: javascript/config.yaml [hunk 4]
```diff
@@ -27,9 +27,9 @@ language:
...
-      command: deno run --allow-net --allow-read --allow-env app.ts
+      command: pm2-runtime start app.ts --interpreter="deno" --interpreter-args="run --allow-net"
     bun:
```
Comment (waghanza) [inline javascript/config.yaml:31]:
```suggestion
    bun:
      deps:
        - npm
      bootstrap:
        -  npm --location=global install pm2
```

**Review by waghanza — CHANGES_REQUESTED** [review]

**Comment by jim-king-2000** [comment]
This error:
```console
curl --retry 5 --retry-delay 5 --retry-max-time 180 --retry-connrefused http://`cat javascript/elysia/ip-bun.txt`:3000 -v
* Closing connection -1
```

It seems that the ip address is not required.

File: javascript/config.yaml [hunk 5]
```diff
@@ -27,9 +27,18 @@ language:
...
+        - npm
+      bootstrap:
+        - npm --location=global install pm2
+      command: pm2 start app.ts --interpreter="deno" --interpreter-args="run --allow-net --allow-read --allow-env" -i max &
     bun:
-      command: bun run app.ts
+      deps:
+        - npm
+      bootstrap:
+        - npm --location=global install pm2
```
Comment (waghanza) [inline javascript/config.yaml:39]:
```suggestion
        - npm -g install pm2
```
not sure why this command does not work

**Review by waghanza — CHANGES_REQUESTED** [review]

**Comment by jim-king-2000** [comment]
Still not OK. The same issue.

File: javascript/config.yaml [hunk 6]
```diff
@@ -27,9 +27,18 @@ language:
...
+        - npm -g install pm2
+      command: pm2 start app.ts --interpreter="deno" --interpreter-args="run --allow-net --allow-read --allow-env" -i max &
     bun:
-      command: bun run app.ts
+      deps:
+        - npm
+      bootstrap:
+        - npm -g install pm2
+        - bun install
+      command: pm2 start "NODE_ENV=production bun app.ts" -i max &
```
Comment (waghanza) [inline javascript/config.yaml:41]:
```suggestion
      command: nohup pm2-runtime start "NODE_ENV=production bun app.ts" -i max
```

**Review by waghanza — CHANGES_REQUESTED** [review]

**Comment by jim-king-2000** [comment]
Good for `bun`, but `deno` still fails.

**Comment by jim-king-2000** [comment]
Accroding to the `bun` startup command, I modify the `deno` startup command as the following:
`nohup pm2-runtime start "deno run --allow-net --allow-read=. app.ts" -i max`
Thank god, it works.

**Comment by jim-king-2000** [comment]
What's wrong with`stricjs`?

**Comment by waghanza** [comment]
I don't know. Any request is failing

Stack is 
```javascript
1 |
2 | const c0=app.static['/']['GET'],c1=app.static['/user']['POST'],f=app.router.find.bind(app.router);return function(r){const s=url.indexOf('/',12) + 1;r.query=r.url.indexOf('?',s);const p=r.query===-1?r.url.substring(s):r.url.substring(s,r.query);switch(p){case'':if(r.method==='GET')return c0(r);break;case'user':if(r.method==='POST')return c1(r);break;}const o=f(r.method,p);if(o===null)return;r.params=o;return o._(r);}
                                                                                                                                                                                                                                                 ^
ReferenceError: Can't find variable: url
```

cc @aquapi

**Comment by jim-king-2000** [comment]
Shall we fix the failure of `stricjs`?

**Comment by waghanza** [comment]
We can try either in this PR or in another, `stricjs` I'd already failing on **master** so both solutions are OK for me.

However, since I cannot publish the results this monday, we can wait a little bit to handle a fix.

**Comment by jim-king-2000** [comment]
OK. I'm going to fix `stricjs`.

**Comment by jim-king-2000** [comment]
The case of `low-http-server` failed. I didn't touch it. So I think this PR is ready to be merged.

**Comment by waghanza** [comment]
Could you merge master please ? @jim-king-2000

**Comment by jim-king-2000** [comment]
> Could you merge master please ? @jim-king-2000

How?

**Review by waghanza — APPROVED** [review]

**Comment by waghanza** [comment]
Will be  done next monday

**Comment by jim-king-2000** [comment]
> Will be done next monday

Could be a little bit earlier?

**Comment by waghanza** [comment]
I can understand the impatience, but no reasons to rush

I prefer to keep the same publishing day (except that I can not this one)

**Comment by jim-king-2000** [comment]
Sure. Take your time.

By the way, I'm just destroyed by this [book](https://www.amazon.com/Move-Fast-Facebook-Builds-Software-ebook/dp/B093HMJ4KB).

**Comment by jim-king-2000** [comment]
> Will be done next monday

No new result?

**Comment by waghanza** [comment]
Something went wrong on my end. Will publish results next week (and ping you)

**Comment by cyrusmsk** [comment]
I hope that it was not my D PRs) looking forward results next week :) thanks @waghanza

**Comment by jim-king-2000** [comment]
> Something went wrong on my end. Will publish results next week (and ping you)

Just a kind reminder, "next week" is coming.

**Comment by waghanza** [comment]
don't worry, it's planned :-)

**Pull request merged**